### PR TITLE
MOD Complexity rebalance

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Reduced the cost of a lot of MODules.

Pathfinder 2 -> 1
Tether 3 -> 1
Temperature Regulator 2 -> 1
DNA lock 2 -> 1
Health analyzer 2 -> 1
Sonar 2 -> 1
Microwave beam 2 -> 1
Drill 2 -> 1
All visors (including NV and thermals) 2 -> 1
Circuit Adapter 2 -> 1

The Mining MODsuit has had its complexity increased to 15 and now starts with the eating apparatus module, with a total base complexity of 10/15 now.

The Prototype MODsuit's active slowdown has been decreased from 1.5 (!) to 1.
## Why It's Good For The Game

> Reduced the cost of a lot of MODules.

There's lots of cute little MODules here, and they are all despite their 'small' cost far too expensive for them to ever be used. The small little cost adds up, when you consider that two 2-complexity modules cost FOUR, which is more than most good modules (that are 3), especially when storage modules take up 3 complexity already. Think about it like genetics, imagine if geladikinesis cost 40 instability. It'd be pointless and just make it not used.

> Pathfinder 2 -> 1

Pathfinder is a little buggy, a bit janky, and still just a commodity, so this might let captains keep it for themselves more often when they're kitting out their MOD.

> Tether 3 -> 1

Tether costing 3 complexity is ABSURD. That's as much as the actual ion jetpacks, and that's for something which you can replace completely with a fire extinguisher, not even including the tiny 4 tiles tethering range.

> Temperature Regulator 2 -> 1

This is vital for spacewalking, I really don't know why it's this expensive. Hell it should be the norm, but whatevs.

> DNA lock 2 -> 1

Nobody's ever going to use this if it can just be EMPed and broken... especially when it costs 2 complexity, which is the same cost as defibs, surgical processor, holster, criminal capture..

> Health analyzer 2 -> 1

This is just a health analyzer. A small item that you're paying for the privilege of being able to have it in your janksuit. It really shouldn't cost 2 complexity, nobody ever takes this.

> Sonar 2 -> 1

I don't think there's much of a reason for sonar to be 2 complexity. You might think it's nuts, but sonar really isn't that useful as it's a windup with a screen-only range. Making it 1 might let it be seen ingame at some point.

> Microwave beam 2 -> 1

Despite the cool name this just fries food. I don't think that should be expensive!

> Drill 2 -> 1

The drill module is mostly redundant when by the time you get it, chances are you have a plasma cutter already which is usually better, if not as space-efficient. There's also the dumb issue with drilling into gibtonite which instantly blows it up.

> All visors (including NV and thermals) 2 -> 1

Similarly to the health analyzer, chances are if you HAVE the module you don't actually *need* it as you're already.. that job. 

Additionally, and this is also part of the reason for the NV, thermal, and even the health analyzer modules, is that traitors/nukies now have to balance MOD economy alongside TC count, and I can't tell you just how frustrating it is to buy something and be told I don't have enough complexity to put it into the MODsuit. I already spent the damn TC!

> Circuit Adapter 2 -> 1

This thing seems pretty useless. All it can really do is open and close your modsuit, which like, wow okay. No need for it to be expensive.

> The Mining MODsuit has had its complexity increased to 15 and now starts with the eating apparatus module, with a total base complexity of 10/13 now.

The complexity increase is because for some reason the MODsuit is already filled to the brim by default, which means that actually interacting with robotics in any way is thoroughly disincentivized as you'd need to take so many modules out to do so that it makes the purchase and interaction pointless. Now you CAN go and ask robotics for anything you need, though there isn't much a miner would want and value enough to trek across the station, for now.

Also, it starts with the eating apparatus because it really looked like it should! The flavor text even talks about miners, it's strange for that to be there if miners won't use it. It'll also encourage it to actually be bought more by allowing you to eat through it.

> The Prototype MODsuit's active slowdown has been decreased from 1.5 (!) to 1.

1.5 is a lot, A LOT, of slowdown. For such an incredibly rare mod, it completely kills the damn thing, even for the charlie station crew! You can't fight xenos with 1.5 slowdown! Having Kinesis isn't enough of a reason to cripple it so thoroughly and pointlessly. It's 0.4 now, which is a nice middleground between 'fast' suits like the medical and security ones, and the 'slow' ones like civilian, engineering, science.
## Changelog
:cl:
balance: Reduced the complexity cost of a lot of MODules.
balance: Pathfinder 2 -> 1
balance: Tether 3 -> 2
balance: Temperature Regulator 2 -> 1
balance: DNA lock 2 -> 1
balance: Health analyzer 2 -> 1
balance: Sonar 2 -> 1
balance: Microwave beam 2 -> 1
balance: Drill 2 -> 1
balance: All visors (including NV and thermals) 2 -> 1
balance: Circuit Adapter 2 -> 1
balance: The Mining MODsuit has had its complexity increased to 13 and now starts with the eating apparatus module, with a total base complexity of 10/13 now.
balance: The Prototype MODsuit's active slowdown has been decreased from 1.5 (!) to 1.
spellcheck: Fixed a type on the energy net module.
/:cl:

### CODE REVIEW

**Comment by moocowswag** [comment]
I would argue that temperature regulator should have a higher cost so people can minmax modsuits for use inside the station rather than in space.

If temperature regulator has a low cost youre literally just shooting yourself in the foot by not taking it because the value of easy spacewalking far outweighs whatever youre gonna get with 1 complexity.

**Comment by DATA-xPUNGED** [comment]
MODsuits bros stay winning

**Comment by TheVekter** [comment]
Yeah, I see no issues with this that moocow hasn't already voiced. Modsuits aren't spaceworthy by default because they'd just outright replace softsuits in every situation. Otherwise these are great, anything to get more people using modsuits on the regular.

**Comment by CRITAWAKETS** [comment]
> I would actually argue that temperature regulator should have a higher cost so people can minmax modsuits for use inside the station rather than in space.
> 
> If temperature regulator has a low cost youre literally just shooting yourself in the foot by not taking it because the value of easy spacewalking far outweighs whatever youre gonna get with 1 complexity and it literally might aswell be an innate module.



> Yeah, I see no issues with this that moocow hasn't already voiced. Modsuits aren't spaceworthy by default because they'd just outright replace softsuits in every situation. Otherwise these are great, anything to get more people using modsuits on the regular.

ok i dont know how to tell this to the two of you but
image
Modsuits are spaceworthy, actually they even work on lizards without the thermal regulation module.

**Comment by TheVekter** [comment]
Wait, so why does the thing even exist?

**Comment by Sealed101** [comment]
plasmemes supposedly

**Comment by Gristlebee** [comment]
> Wait, so why does the thing even exist?

The mining modsuit isn't protected from the cold of space, so it needs a thermal regulator to be proplerly space worthy.

For the other types of suits, yeah in general they don't need the regulator to space walk if your temperature is already stable beforehand which logically you'd think you'd need it for.

Still it's useful in a few other cases, like if you're already cold before engaging the suit.  Or actively being made cold from sources like the temp gun, void heretics, cryo sting and having insulating clothing on might be fatal due to keeping in the cold, or being set on fire and needing to cool off.

**Comment by Likteer** [comment]
>  Modsuits aren't spaceworthy by default because they'd just outright replace softsuits in every situation.

@TheVekter But THEY ARE supposed to be spaceworthy, because they all REPLACED spaceworthy hardsuits.
Cargo MODsuit is a notable exception, it was designed to be something else.

**Comment by Serxule** [comment]
fun fact: temp module only makes any difference with heat/nonspace-proof modsuits, since all spaceproof modsuits are fully cold resistant by default (space being cold and such)

**Comment by Fikou** [comment]
> Wait, so why does the thing even exist?

lizards

File: code/modules/mod/modules/modules_medical.dm [hunk 1]
```diff
@@ -8,12 +8,12 @@
 /obj/item/mod/module/health_analyzer
 	name = "MOD health analyzer module"
 	desc = "A module installed into the glove of the suit. This is a high-tech biological scanning suite, \
-		allowing the user indepth information on the vitals and injuries of others even at a distance, \
+		allowing the user indepth information on the vitals and injuries of others even at a distance, including even chemical scans, \
```
Comment (Fikou) [inline code/modules/mod/modules/modules_medical.dm:11]: even twice is weird wording

File: code/modules/mod/mod_theme.dm [hunk 2]
```diff
@@ -1302,8 +1302,8 @@
 	siemens_coefficient = 0
 	complexity_max = DEFAULT_MAX_COMPLEXITY + 5
 	charge_drain = DEFAULT_CHARGE_DRAIN * 2
-	slowdown_inactive = 2
-	slowdown_active = 1.5
+	slowdown_inactive = 3
+	slowdown_active = 0.4
```
Comment (Fikou) [inline code/modules/mod/mod_theme.dm:1306]: nah, at max make it as fast as the normal engineering ones

File: code/modules/mod/modules/modules_engineering.dm [hunk 3]
```diff
@@ -85,7 +85,7 @@
 		these are only capable of working in zero-gravity environments, a blessing to some Engineers."
 	icon_state = "tether"
 	module_type = MODULE_ACTIVE
-	complexity = 3
+	complexity = 1
```
Comment (Fikou) [inline code/modules/mod/modules/modules_engineering.dm:88]: keep this at 2

File: code/modules/mod/mod_theme.dm [hunk 4]
```diff
@@ -317,7 +317,7 @@
 	resistance_flags = FIRE_PROOF|LAVA_PROOF
 	max_heat_protection_temperature = FIRE_SUIT_MAX_TEMP_PROTECT
 	min_cold_protection_temperature = FIRE_SUIT_MIN_TEMP_PROTECT
-	complexity_max = DEFAULT_MAX_COMPLEXITY - 5
+	complexity_max = DEFAULT_MAX_COMPLEXITY
```
Comment (Fikou) [inline code/modules/mod/mod_theme.dm:320]: you already lower the module value of a bunch of the modules this uses (and useful sidegrades for it like the thermal regulator). you aren't meant to be able to put everything into every suit.

**Review by Fikou — CHANGES_REQUESTED** [review]

File: code/modules/mod/modules/modules_engineering.dm (hunk 3 above)
Comment (carlarctg) [reply code/modules/mod/modules/modules_engineering.dm:88]: Why? This is literally an inbuilt fire extinguisher dude. It doesn't even reach as far as one.

File: code/modules/mod/mod_theme.dm (hunk 2 above)
Comment (carlarctg) [reply code/modules/mod/mod_theme.dm:1306]: Gay

File: code/modules/mod/mod_theme.dm (hunk 4 above)
Comment (carlarctg) [reply code/modules/mod/mod_theme.dm:320]: The only thing I lowered the value of was the drill module. In total, it still costs 9 complexity. My arguments still hold. 

The mining modsuit is underused because everything it has is a sidegrade of pre-existing mining gear, while locking you into it and making you unable to get many of the lavaland items. 

Making it utterly restricted in any modules to actually upgrade the frickin' thing just means there is NO reason to waste 2k credits on what amounts to worse drake armor, worse plasma cutter, awkward bag of holding, and a drastic reduction in storage size. It's useless by itself. 

Letting it actually GET valuable upgrades without cannibalizing it to the point you have no reason to use it gives it one small reason to be picked over normal lavaland equipment.

'You arent meant to put everything into one suit' doesn't hold because the everything that's in the suit is completely redundant, and by removing those you discard the reasons to purchase the thing to begin with.

**Comment by CoiledLamb** [comment]
as someone who's bought the mining modsuit several times it is severely unloved at present, locking the player out of using even useful mining vendor items like kheiral cuffs, and roundstart items like their boot storage. it's the kind of item that seems appealing at a glance, but actually playing with it proves that it's an item that will get you killed in several cases, and prevent players from finding your corpse. all in all i think these changes are great, as whenever i would get it upgraded from the roboticist, i would end up throwing away more than half the mods inside as they have no practical use when compared to items that you get at roundstart that are incredibly easy to store. after a while i just started asking for expanded storage only because it wasn't worth the effort to completely gut the thing, and shortly after that i shelved it altogether.
tl;dr the mining modsuit clearly has potential, but it's held back by starting with exclusive mining mods that miners don't even really want, and can't excel because it is constantly at odds with useful utility items you get from the vendor or lavaland.

**Comment by Exester509** [comment]
> plasmemes supposedly

Right, the THERMAL regulator exists for TEMPERATURE IMMUNE plasmemes. They already have their own niche module, so I assumed it was an analog for lizards

File: code/modules/mod/modules/modules_engineering.dm (hunk 3 above)
Comment (Fikou) [reply code/modules/mod/modules/modules_engineering.dm:88]: what does this even mean, it's longer than your view range both horizontally and vertically

File: code/modules/mod/mod_theme.dm (hunk 2 above)
Comment (Fikou) [reply code/modules/mod/mod_theme.dm:1306]: ok

Comment (Fikou) [reply code/modules/mod/mod_theme.dm:1306]: happy pride month

File: code/modules/mod/modules/modules_engineering.dm (hunk 3 above)
Comment (DATA-xPUNGED) [reply code/modules/mod/modules/modules_engineering.dm:88]: Feels like this module will pretty much always be quite useless as long as we can still grab one credit out of the ID and use that as propulsion

**Comment by necromanceranne** [comment]
> Wait, so why does the thing even exist?

It stabilizes temperature in both directions. So if you're cooled or heated by an external effect (like, for instance, being on fire or hit with a temperature gun, or maybe in proximity to a fusion fire), it will start to correct your temperature.

While modsuits have temperature protections, they don't regulate your temperature if an effect would bypass those protections. This is important for protecting you in situations where you are being exposed to extreme temperature effects. Or, yes, for species that lack innate temperature stabilization. I've used it during freon fires.

File: code/modules/mod/modules/modules_engineering.dm (hunk 3 above)
Comment (necromanceranne) [reply code/modules/mod/modules/modules_engineering.dm:88]: I'm not fumbling for credits if I could just use the tether to jettison myself along during zero grav. The complexity definitely is the real holding this module back, it has value once you start seeing some of the neat combinations it has with other modules.

File: code/modules/mod/mod_theme.dm (hunk 4 above)
Comment (necromanceranne) [reply code/modules/mod/mod_theme.dm:320]: I think at this point I have to agree that the mining modsuit doesn't quite feel like a value purchase unless topside has completely stagnated in research for some reason, which is absolutely worst case scenario. While it technically does provide extra storage in that it saves a few pieces of equipment in pockets/your bag (namely the GPS, ore satchel and pickaxe/shovel), as well as carrying objects in crates (but not bodies obviously), you can't upgrade it with other modules at all. That limitation is especially harsh given the poor EVA capabilities.

I absolutely agree that since I have to cannibalize the suit to give it alternative functionality for some reason or another, I would ultimately be better off using a different modsuit from robotics and sticking with my standard gear and suits. It's only main draw at that point is the potential for bigger bag space, something capable even with a standard modsuit,  which also gives me inbuilt EVA. And by doing so, I can fit the modsuit with more modules overall, becoming more flexible as a result. A very mild slowdown from a standard suit hardly matters.

The mining bomb is neat but gimmicky, and majorly slower than equivalent mining gear at the point you can buy the suit. I did like the lava traversal, but not enough for me to stick to the suit. Everything else but the ash armor is replicable with any modsuit.

File: code/modules/mod/modules/modules_medical.dm (hunk 1 above)
Comment (carlarctg) [reply code/modules/mod/modules/modules_medical.dm:11]: Apparently chem scans are also on normie analyzers so the change is going away
